### ISSUE TITLE
Podmin should be able to forward a report to the source pod podmin

### ISSUE THREAD

**Issue by Flaburgan** [diaspora/diaspora#7316]
When a post from another pod is reported to us, we podmins often want to warn the source pod podmin about it. At the moment this is impossible inside diaspora we have to find the podmin to manually contact him. I would like to add a button to be able to forward the report to the source pod. The report would be sent through the federation and the pod receiving it will then locally process tu as a traditionnal report (warning all admins and moderators).

**Comment by dmorley** [comment]
Would be cool if the report somehow made it automatically to the right podmin, 99% of the reports we get are items we can do nothing with.

**Comment by SuperTux88** [comment]
If the reported item is a comment, it could also be sent to the author of the post, because they can delete comments from the post. That way it takes work from the podmins.

**Comment by denschub** [comment]
> it could also be sent to the author of the post

Which would require us to create a report UI for all users. Sounds pretty awesome, but also sounds like a pretty big task.

**Comment by Flaburgan** [comment]
>Would be cool if the report somehow made it automatically to the right podmin, 99% of the reports we get are items we can do nothing with.

Well the point is, pods are hosted everywhere and the laws vary. So a post which is illegal at the pod where it is reported is maybe legal at the source pod. So it is nice to be warned and maybe to remove the post without depending of the "master" pod.

Same remark for @SuperTux88 suggestion, the podmin is the legal responsible, so he has to be warned anyway, if the author doesn't moderate the comment, the podmin can be in trouble.

**Comment by SuperTux88** [comment]
When it uses federation-code, it is pretty easy to add multiple recipients (local or remote), so we can send it to the local podmin, remote podmin and post author. A lot of reports (99%) are "mimimi, he said something rude to me :(", so it doesn't matter for me as a podmin, but maybe the post author wants to delete it. Or if it is something about law, maybe the post author is faster with deletion than the podmin, or if it is spam, the local podmin deletes it already for the local users, but the remote-podmin can delete it for everybody later. So it can be useful to have multiple recipients anyway, and as said, it is easy to add them in the federation system.

The only thing we need to do is to build (or reuse) the UI for all users.

**Comment by goobertron** [comment]
> we can send it to the local podmin, remote podmin and post author.

I can see some possible advantages from including the post's author in report recipients, such as those you mention, but I can also see the possibility of abuse/trolling. We certainly wouldn't want to include the name of the reporter to the post author, as that could spark abuse; on the other hand, would we want to show the report text to the post author? That could equally be used for abuse.

Perhaps a better solution might be a button to allow the podmin to forward the report text to the post author if they think it would be helpful, rather than including the author in the recipients automatically.

**Comment by SuperTux88** [comment]
I don't know why we shouldn't send the name and text to the author, because the author is basically moderator of their own posts. And they should know the name so they can block people who abuse the report feature and spam with reports and it's also important to see the report-text.

**Comment by goobertron** [comment]
I think it would be very off-putting to many people in reporting certain types of more extremist content. Consider:

- a woman reporting violent misogynistic content;
- someone from an ethnic minority reporting violently racist content, including a Jewish person reporting neo-Nazi content;
- someone from the LGBTQ community reporting violently homophobic content.

Just consider what has happened to women who have even mildy criticised misogynistic content on platforms such as Twitter, or to those from the other groups. Up to an including death threats.

There will be many other cases. I just don't think we'd be doing anything to protect our community by showing the report to the person reported. In some cases that would be helpful, but there's no way to decide whether or not it would be appropriate using software, so let the podmin decide. And never send the reporter's name to the post author, just in case either the podmin is mischievous (or shares the extreme views of the author) or clicks the wrong button. A person reported doesn't need to know who is reporting them. If someone is abusing the report system, the podmin can deal with it.

Another danger with sending the report automatically to the post author would be the converse of the above cases; racist, homophobic or misogynist trolls could use it to send in effect private messages to anyone posting feminist, queer or racially liberal content (apologies if any of my terms aren't quite correct). Particularly if someone is able to report posts made by someone who has ignored them (which I think should be possible, because there will be cases where it will be necessary), that can be abused. Even if a podmin steps in to stop it, a lot of distress and harm could be caused before the podmin even saw the reports.

**Comment by SuperTux88** [comment]
> I just don't think we'd be doing anything to protect our community by showing the report to the person reported.

No, not to the reported person, to the author of the post. So for example Person A creates a post, person B comments something to the post, person C then reports this comment and the report would also be sent to the author of the post (person A), but person B would never see the report. But I think person A needs this information.

That would only happen when comments are reported, when posts are reported the reports are only sent to the podmin.

> A person reported doesn't need to know who is reporting them.

I fully agree with this!

**Comment by tyzoid** [comment]
> No, not to the reported person, to the author of the post. So for example Person A creates a post, person B comments something to the post, person C then reports this comment and the report would also be sent to the author of the post (person A), but person B would never see the report. But I think person A needs this information.

What should happen, in your opinion, if person A posts a comment to their own post (responding to B), and C reports A's comment?

**Comment by SuperTux88** [comment]
> What should happen, in your opinion, if person A posts a comment to their own post (responding to B), and C reports A's comment?

Then only the podmin should receive the report. (As said, the reported person shouldn't receive the report)

> Plus, Person A could be posting from an alternate account (thus B = A), or could have partially incited B's comment.

Person B could also be the podmin ... or Person A could be the podmin (Yes, people on my pod already reported posts/comments from me to me ...)

> Because of all the weird edge cases, I think that it would be a good idea to not send the report to A at all.

Because of all the weird edge cases we should probably remove the report feature completely ... only that way we can be sure to never send report-messages to anybody ...

**Comment by svbergerem** [comment]
> But I think person A needs this information.

No, I don't think so. Users shouldn't have to deal with reports written about other users.

> A person reported doesn't need to know who is reporting them.

And neither does their podmin. (Unless all involved users are on the same pod.)

When a user reports a post/comment/whatever, I think the podmin should have to following options:

1. For reported content that came from the same pod:
  a) delete the content, resolves the reported issue.
  b) send the author of the content a (possibly pre-formulated) private message (using our conversations feature)
  c) lock/delete the user's account, deleting resolves the reported issue

  a)+b) should be possible from the report view

2. For reported content that came from another pod:
  a) Send a report to the podmin from the other pod. The text of the report should be pre-formulated (e.g. by copying the original report text), but the podmin should be able to change it and remove confidential information. The author of the original report shouldn't be mentioned.
  b) delete the content, resolves the reported issue
  c) delete+ban the user from the pod, resolves the reported issue

Podmins can handle users abusing the report feature by sending them a private message and locking/deleting the user's account if that doesn't help. We can think about allowing podmins to blacklist other pods so they a) don't receive reports from this pod b) don't receive any content from this pod.

**Comment by goobertron** [comment]
> Person B could also be the podmin ... or Person A could be the podmin (Yes, people on my pod already reported posts/comments from me to me ...)

Good point; but we already tell users that they have to trust their podmin with their data. Perhaps in that case we shouldn't send the identity of the person who made the report to the remote podmin when the post/comment originates from a remote pod. The remote podmin should receive only the report content.

**Comment by tyzoid** [comment]
> Perhaps in that case we shouldn't send the identity of the person who made the report to the remote podmin when the post/comment originates from a remote pod. The remote podmin should receive only the report content

I think that it might be useful to include the pod that the report originated from, which would allow the remote podmin to contact the podmin if there's an abuse of the report system.

**Comment by Flaburgan** [comment]
IMO, the podmins are the ones who are legally responsible. So they are the ones who should receive the report. I don't see the point to send the report for a comment to the post authors. They probably already received a notification about a new comment, if they choose not to delete it at the moment, they will probably not do it after a report.

So back to the original topic: podmins should be able to forward a report to the source pod podmin ;)

**Comment by SuperTux88** [comment]
> They probably already received a notification about a new comment

Ok, since they can't unsubscribe notifications for an own post, that's probably true, so I think it's fine then to not send reports to them.

> So back to the original topic: podmins should be able to forward a report to the source pod podmin ;)

I think this should happen automatically immediately, or let the reporter choose which podmins (own and/or source podmin) should receive it, if a source podmin is known. That makes it much faster and don't add an extra delay for the own podmin to react to it and forward it (some podmins are very inactive).

**Comment by Flaburgan** [comment]
@SuperTux88 so the federation protocol should be modified to allow that right? Should I open an issue on that repository then?

**Comment by SuperTux88** [comment]
Yes, it would need a `report` entity or something like that.

**Comment by shawneric** [comment]
It makes sense to me that the person reporting should have the option to include the initial poster's name or not. Sometimes I'd like to report someone and stay anonymous. In fact, most reports should be this way, as you don't want to somehow make yourself end up being a target (which is seen quite often these days on social media networks). 

Also, I know when I get a report about my own post, I become defensive immediately, as do most people and feel the need to defend myself against what I pressume is an attack. Of course it isn't, but this is a natural human response. It's better to allow a site administrator to individually speak with that person and just say "we have had several reports about your posts by members in the federated community." etc. etc.

**Comment by MasterOfTheTiger** [comment]
We really should have the option to report without it being forwarded. For example, my pod deletes a lot of content a remote podmin likely would be fine having.
